From d47d281214a4393197feb7a3f53c6915d26b9d6b Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Thu, 17 Sep 2020 10:25:30 +0800
Subject: [PATCH] Revert "f2fs: fix memory leak in
 f2fs_try_convert_inline_dir()"

This reverts commit a8baa34ace934e04c399546615038ac939884f5a.

Replaced by commit 62260dfc3b27 ("f2fs: don't leak filename in f2fs_try_convert_inline_dir()").

Signed-off-by: Tao Huang <huangtao@rock-chips.com>
Change-Id: I9b9e7c8278d40a42c679b9dd08b26c7a0b107b83
---
 fs/f2fs/inline.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/fs/f2fs/inline.c b/fs/f2fs/inline.c
index eb500dd0ff7b..e85a005c7b7f 100644
--- a/fs/f2fs/inline.c
+++ b/fs/f2fs/inline.c
@@ -579,7 +579,7 @@ int f2fs_try_convert_inline_dir(struct inode *dir, struct dentry *dentry)
 
 	err = f2fs_setup_filename(dir, &dentry->d_name, 0, &fname);
 	if (err)
-		goto out_unlock;
+		goto out;
 
 	ipage = f2fs_get_node_page(sbi, dir->i_ino);
 	if (IS_ERR(ipage)) {
@@ -600,8 +600,6 @@ int f2fs_try_convert_inline_dir(struct inode *dir, struct dentry *dentry)
 out_fname:
 	f2fs_free_filename(&fname);
 out:
-	fscrypt_free_filename(&fname);
-out_unlock:
 	f2fs_unlock_op(sbi);
 	return err;
 }
-- 
2.35.3

